lambeq/rewrite/base.py

```diff
@@ -393,3 +399,16 @@ def _ar(self, box: Box) -> Diagram:
 
     def _ob(self, ob: Ty) -> Ty:
         return ob
+
+
+class ImperativeRewriteRule():
+    """A rewrite rule for imperative sentence diagrams.
+    """
+    def matches(self, diagram: Diagram) -> bool:
+        return diagram.cod == N.r @ S
+
+    def rewrite(self, diagram: Diagram) -> Diagram:
+        if self.matches(diagram):
+            return (Box('IMP', Ty(), N) @ diagram >> Cup(N, N.r) @ Id(S))
+        else:
+            raise ValueError(f'The diagram codomain {diagram.cod} was passed, but N.r @ S was expected.')
```

Comment:
A rewrite rule should not raise an error if the condition is not matched. Just return the original diagram here.